src/core/vector/qgsvectorlayer.cpp

```diff
@@ -874,7 +874,7 @@ QgsRectangle QgsVectorLayer::extent() const
   }
 
   if ( !mEditBuffer ||
-       ( mEditBuffer->mDeletedFeatureIds.isEmpty() && mEditBuffer->mChangedGeometries.isEmpty() ) ||
+       ( !mDataProvider->transaction() && ( mEditBuffer->deletedFeatureIds().isEmpty() && mEditBuffer->changedGeometries().isEmpty() ) ) ||
```

Comment:
It would be a bit cleaner if the !mDataProvider->transaction() checks were instead handled by either dynamic casting mEditBuffer to QgsVectorLayerEditPassthrough or by a new "isPassthrough" or "buffersChanges" method QgsVectorLayerEditBuffer which is overridden by QgsVectorLayerEditPassthrough. Then we'd have a cleaner separation between transactions and the use of QgsVectorLayerEditPassthrough (which may potentially be used in other situations in future, or by plugins...)

Reply:
I've added `isPassthrough()` to edit buffer for a clearer semantic